packages/koenig-lexical/src/components/ui/LabelDropdown.jsx

```diff
@@ -0,0 +1,144 @@
+        return (
+            <Item key={item.name} item={item} selected={selected} onChange={handleSelect}/>
+        );
+    };
+
+    const allLabels = menu.concat(newLabels).map(label => ({...label, label: label.name}));
+    const [selectedLabels, nonSelectedLabels] = partition(allLabels, label => value?.includes(label.id));
+    const filteredItems = nonSelectedLabels.filter(item => item.name.toLowerCase().includes(filter.toLowerCase()));
+    const emptyItem = filter && !selectedLabels?.some(label => label.name === filter)
+        ? [{id: undefined, name: filter, label: <>Add <strong>&quot;{filter}&quot;...</strong></>}]
+        : [{id: undefined, name: undefined, label: 'Type to search'}];
+
+    return (
+        <div className="relative font-sans text-sm font-normal" data-testid={dataTestId}>
+            <div
+                className={`border-grey-300 text-grey-900 dark:border-grey-900 dark:bg-grey-900 dark:placeholder:text-grey-800 relative flex w-full cursor-text flex-wrap gap-1 border py-2 pl-3 pr-5 text-left font-sans font-normal focus-visible:outline-none dark:text-white ${open ? 'rounded-t' : 'rounded'}`}
+                type="button"
+                onClick={() => inputRef.current.focus()}
+            >
+                {selectedLabels.map(label => (
+                    <button
+                        key={label.id}
+                        className="bg-grey-900 dark:bg-grey-100 dark:text-grey-900 flex cursor-pointer items-center rounded-sm py-1 px-2 leading-none text-white"
+                        data-testid="label-dropdown-selected"
+                        type="button"
+                        onMouseDownCapture={event => handleDeselect(event, label)}
+                    >
+                        {label.label}
+                        <CloseIcon className="ml-2 h-2 w-2" />
+                    </button>
+                ))}
+
+                <div className="flex-1">
+                    <input
+                        ref={inputRef}
+                        className="h-full w-full min-w-[5rem] appearance-none bg-transparent px-1 leading-none outline-none"
+                        placeholder=""
+                        value={filter}
+                        onBlur={handleBlur}
+                        onChange={event => setFilter(event.target.value)}
+                        onFocus={handleOpen}
+                        onKeyDown={handleBackspace}
+                    />
+                </div>
+
+                <ArrowIcon className={`text-grey-600 absolute right-2 top-4 h-2 w-2 ${open && 'rotate-180'}`} />
+            </div>
+            {open && (
+                <DropdownContainer>
+                    <KeyboardSelection
+                        getItem={getItem}
+                        items={filteredItems.length ? filteredItems : emptyItem}
+                        onSelect={handleSelect}
+                    />
+                </DropdownContainer>
+            )}
+        </div>
+    );
+}
```

Comment:
Awesome! Would suggest renaming this component (and some of the variable names) to something more generic, like MultiselectDropdown / items instead of "labels"